staging/src/k8s.io/apiserver/pkg/storage/testing/store_tests.go

```diff
@@ -619,6 +621,248 @@ func RunTestPreconditionalDeleteWithOnlySuggestionPass(ctx context.Context, t *t
 	expectNoDiff(t, "incorrect pod:", updatedPod, out)
 }
 
+func RunBenchmarkStoreListCreate(ctx context.Context, b *testing.B, store storage.Interface, match metav1.ResourceVersionMatch) {
+	objectCount := atomic.Uint64{}
+	pods := []*example.Pod{}
+	for i := 0; i < b.N; i++ {
+		name := randomString(100)
+		pods = append(pods, &example.Pod{ObjectMeta: metav1.ObjectMeta{Namespace: "ns", Name: name}})
+	}
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		pod := pods[i]
+		podOut := &example.Pod{}
+		err := store.Create(ctx, computePodKey(pod), pod, podOut, 0)
+		if err != nil {
+			panic(fmt.Sprintf("Unexpected error %s", err))
+		}
+		listOut := &example.PodList{}
+		err = store.GetList(ctx, "/pods", storage.ListOptions{
+			Recursive:            true,
+			ResourceVersion:      podOut.ResourceVersion,
+			ResourceVersionMatch: match,
+			Predicate: storage.SelectionPredicate{
+				Label: labels.Everything(),
+				Field: fields.Everything(),
+				Limit: 1,
+			},
+		}, listOut)
+		if err != nil {
+			panic(fmt.Sprintf("Unexpected error %s", err))
+		}
+		objectCount.Add(uint64(len(listOut.Items)))
```

Comment:
nit: let's also check that we indeed returned `i+1` items

Reply:
Added